src/capwap_http_api_handler.erl

```diff
@@ -20,408 +20,459 @@
 
 handle_request_json(Req, State) ->
-    Path = binary:split(cowboy_req:path(Req), <<"/">>, [global, trim_all]),
-    Method = cowboy_req:method(Req),
-    case Path of
-        [<<"api">>, <<"v1">>, <<"wtp">> | TailPath] ->
-            handle_request_wtp(Method, TailPath, Req, State);
-        [<<"api">>, <<"v1">>, <<"dp">> | TailPath] ->
-            handle_request_dp(Method, TailPath, Req, State);
-        [<<"api">>, <<"v1">>, <<"station">> | TailPath] ->
-            handle_request_station(Method, TailPath, Req, State);
-        [<<"api">>, <<"v1">>, <<"version">> ] ->
-            case Method of
-                <<"GET">> ->
-                    {get_release_vsn(), Req, State};
-                _ ->
-                    {jsx:encode([{error, bad_command}]), Req, State}
-            end;
-        [<<"metrics">> | TailPath] ->
-            TailPath1 = [ binary_to_existing_atom(P, utf8) || P <- TailPath ],
-            handle_request_metrics(json, TailPath1, Req, State);
-        _ ->
-            {jsx:encode([{error, bad_command}]), Req, State}
-    end.
+	Path = binary:split(cowboy_req:path(Req), <<"/">>, [global, trim_all]),
+	Method = cowboy_req:method(Req),
+	case Path of
+		[<<"api">>, <<"v1">>, <<"wtp">> | TailPath] ->
+			handle_request_wtp(Method, TailPath, Req, State);
+		[<<"api">>, <<"v1">>, <<"dp">> | TailPath] ->
+			handle_request_dp(Method, TailPath, Req, State);
+		[<<"api">>, <<"v1">>, <<"station">> | TailPath] ->
+			handle_request_station(Method, TailPath, Req, State);
+		[<<"api">>, <<"v1">>, <<"version">> ] ->
+			case Method of
+				<<"GET">> ->
+					{get_release_vsn(), Req, State};
+				_ ->
+					{jsx:encode([{error, bad_command}]), Req, State}
+			end;
+		[<<"api">>, <<"v2">> | TailPath ] ->
+			capwap_http_api_v2_handler:handle(Method, TailPath, Req, State);
+		[<<"metrics">> | TailPath] ->
+			TailPath1 = [ binary_to_existing_atom(P, utf8) || P <- TailPath ],
+			handle_request_metrics(json, TailPath1, Req, State);
+		_ ->
+			{jsx:encode([{error, bad_command}]), Req, State}
+	end.
 
 handle_request_text(Req, State) ->
-    Path = binary:split(cowboy_req:path(Req), <<"/">>, [global, trim_all]),
-    Method = cowboy_req:method(Req),
-    case {Method, Path} of
-        {<<"GET">>, [<<"metrics">> | TailPath] } ->
-            TailPath1 = [ binary_to_existing_atom(P, utf8) || P <- TailPath ],
-            handle_request_metrics(text, TailPath1, Req, State);
-        _ ->
-            {<<"error: bad_command">>, Req, State}
-    end.
+	Path = binary:split(cowboy_req:path(Req), <<"/">>, [global, trim_all]),
+	Method = cowboy_req:method(Req),
+	case {Method, Path} of
+		{<<"GET">>, [<<"metrics">> | TailPath] } ->
+			TailPath1 = [ binary_to_existing_atom(P, utf8) || P <- TailPath ],
+			handle_request_metrics(text, TailPath1, Req, State);
+		_ ->
+			{<<"error: bad_command">>, Req, State}
+	end.
+
+delete_resource(Req, State) ->
+	handle_request_json(Req, State).
+
+resource_exists(Req = #{bindings := Vars}, State) ->
```

Comment:
this function make my head spin, drop the fold and use a recursive function instead.

Reply:
this was not resolved, the logic in this function is still expressed in a way that is way to complicated.

HINT: If you have more that 2 level of indented block, your should start thinking about rewriting. Beyond 4 it almost always should be rewritten (here we have foldl->fun->case->case ... that is definitely to much to make sense).